### ISSUE TITLE
Invalid value additional_image_links

### ISSUE THREAD

**Issue by smazur** [woocommerce/google-listings-and-ads#1097]
### Describe the bug:

I found following message in log and some of our products do not sync to merchant center because of this error.

```
2021-11-04T15:27:55+00:00 ERROR Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductSyncer::update_by_batch_requests {
  "error": {
    "code": 400,
    "message": "Invalid value at 'body.entries[0].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[1].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[2].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[3].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[4].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[5].product' (additional_image_links), Starting an object on a scalar field",
    "errors": [
      {
        "message": "Invalid value at 'body.entries[0].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[1].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[2].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[3].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[4].product' (additional_image_links), Starting an object on a scalar field\nInvalid value at 'body.entries[5].product' (additional_image_links), Starting an object on a scalar field",
        "reason": "invalid"
      }
    ],
    "status": "INVALID_ARGUMENT"
  }
}
```

### Expected behavior:
Products with multiple images should properly sync.

### Actual behavior:
Products with multiple images do not sync.

### Additional details:
After applying following code the issue disappeared.

```
function fix_additional_image_links( $values, $wc_product, $wc_product_adapter ) {
    $additional_image_links = $wc_product_adapter->getAdditionalImageLinks();

    if( is_array( $additional_image_links ) ) {
        $values['additionalImageLinks'] = implode(',', array_map( static function( $x ) {
            return str_replace(',', '%2C', $x );
        }, $additional_image_links ) );
    }

    return $values;
}

add_filter('woocommerce_gla_product_attribute_values', 'fix_additional_image_links', 10, 3);

```

Also need to change additionalImageLinks attribute validation in WCProductAdapter class to following:

```
$metadata->addPropertyConstraint( 'additionalImageLinks', new Assert\Length( null, 0, 2000 ) );
```

**Comment by puntope** [comment]
Hey there @smazur 

Thanks for using Google Listings & Ads!

We were unable to reproduce the issue with additional_image_links. Could be maybe an outdated version or a compatibility plugin I would say. Anyway, I saw you solved it with some hooks. 

Given that, I wanted to point out that this repository is not intended for support requests or opening tickets. The best place to start the process is with the [WordPress.org Google Listings and Ads forum](https://wordpress.org/support/plugin/google-listings-and-ads/) or if you have a WooCommerce.com account, you can start a chat or open a ticket on [WooCommerce.com support](https://woocommerce.com/my-account/create-a-ticket/). 

The support team will help guide you through some problem solving and get some more information to give us a bit more context/background on your concerns/report above.

**Comment by smazur** [comment]
Hi @puntope 

Your reply made me get back to investigate our problem deeper. I found that fix solution provided above is wrong. Also I found that `$wc_product->get_gallery_image_ids()` returns 1-based array of ids and that causes `json_encode` to convert php array to json object and additional_image_links validation failure. Consider following code:

```
json_encode([1=>1,2,3]); // Produces {"1":1,"2":2,"3":3} instead of [1,2,3]
```

We have filter that excludes featured image id from gallery image ids array to avoid image duplication. That filter produces this 1-based array. 

It would be nice to wrap all array attributes into `array_values` function to avoid this problem.
